src/sys/ptrace.rs

```diff
@@ -140,3 +150,110 @@ pub fn ptrace_setsiginfo(pid: Pid, sig: &siginfo_t) -> Result<()> {
         Err(e) => Err(e),
     }
 }
+
+//-------------------------- Second part: a high-level wrapper for ptrace ----------------------//
+
+#[cfg(target_arch = "x86_64")]
+// We're going to export it anyway
+#[allow(dead_code)]
+#[allow(non_camel_case_types)]
+pub enum Register {
+    R15 = 0 * 8,
+    R14 = 1 * 8,
+    R13 = 2 * 8,
+    R12 = 3 * 8,
+    RBP = 4 * 8,
+    RBX = 5 * 8,
+    R11 = 6 * 8,
+    R10 = 7 * 8,
+    R9 = 8 * 8,
+    R8 = 9 * 8,
+    RAX = 10 * 8,
+    RCX = 11 * 8,
+    RDX = 12 * 8,
+    RSI = 13 * 8,
+    RDI = 14 * 8,
+    ORIG_RAX = 15 * 8,
+    RIP = 16 * 8,
+    CS = 17 * 8,
+    EFLAGS = 18 * 8,
+    RSP = 19 * 8,
+    SS = 20 * 8,
+    FS_BASE = 21 * 8,
+    GS_BASE = 22 * 8,
+    DS = 23 * 8,
+    ES = 24 * 8,
+    FS = 25 * 8,
+    GS = 26 * 8,
+}
+
+type Word = usize;
+
+/// Makes the `PTRACE_SYSCALL` request to ptrace
+pub fn syscall(pid: Pid) -> Result<()> {
```

Comment:
This should be called `ptrace_syscall`, we're prefixing our helper functions with `ptrace_` to make it more clear. `syscall` is already a POSIX function, so we don't want any confusion here.